### PR TITLE
Remove shims

### PR DESCRIPTION
**Pull request by ptomato**
Removes ad-hoc shims and workarounds for Symbol, WeakMap, and Object.assign.

Closes: #449.

### CODE REVIEW

File: polyfill/lib/ecmascript.mjs [hunk 1]
```diff
@@ -1354,7 +1353,18 @@ function bisect(getState, left, right, lstate = getState(left), rstate = getStat
   }
   return right;
 }
-
+function unique(arr) {
```
Comment (ljharb) [inline polyfill/lib/ecmascript.mjs]: This whole function could be `return [...new Set(arr)]`

File: polyfill/lib/intl.mjs [hunk 2]
```diff
@@ -118,7 +117,7 @@ function amend(options = {}, amended = {}) {
 function timeAmend(options) {
   options = amend(options, { year: false, month: false, day: false });
   if (!hasTimeOptions(options)) {
-    options = ObjectAssign(options, {
+    options = Object.assign(options, {
```
Comment (ljharb) [inline polyfill/lib/intl.mjs]: Note that if you want these to be robust against runtime env modification, you still have to cache the function at module level.

File: polyfill/lib/ecmascript.mjs (hunk 1 above)
Comment (ptomato) [reply polyfill/lib/ecmascript.mjs]: Unfortunately, that doesn't seem to work with the wrappers from the big-integer library.

Comment (ljharb) [reply polyfill/lib/ecmascript.mjs]: ah, because they don't use `===`, gotcha.

File: polyfill/lib/ecmascript.mjs [hunk 3]
```diff
@@ -1355,7 +1354,18 @@ function bisect(getState, left, right, lstate = getState(left), rstate = getStat
...
+  for (var i = 0; i < arr.length; i++) {
+    var v = arr[i];
+    obj[v] = v;
+  }
+  var res = [];
+  for (var p in obj) {
+    res.push(obj[p]);
+  }
+  return res;
+}
```
Comment (Ms2ger) [inline polyfill/lib/ecmascript.mjs]: This is fine, but pretty heavy-handed for one call with an array of two elements. Followup to remove entirely?

Comment (ptomato) [reply polyfill/lib/ecmascript.mjs]: :+1:, see #488.

**Pull request merged**